### DISCUSSION TITLE
Laravel 9 Migration How To Set Date Default Value ?

### DISCUSSION THREAD

**Discussion by bhargavraviya** [laravel/framework#45937] [Q&A]
In Laravel Migration how to set the date default value?

$table->date('date')->default(?);

**Comment by henzeb** [comment]
You want an actual static date to be default, or now on create / update?

**Comment by denizgolbas** [comment]
If you catch current date for record new line on database you can use observers or mutators in model. Or you can set function with DB:statement in your migration up method. But best is with mutators.

**Reply by bhargavraviya** [reply]
your right but it is not the perfect answer to make the default 
This way proper to set the date default value by laravel. not a migration

**Reply by denizgolbas** [reply]
```php
$table->timestamp(date');
```

This will insert data type is timestamp and default=current_timestamp

If you want data type is DATE and default=current_timestamp it is not possible for mysql. You need to use sql functions and you need to set default=thecurrentdatefunction.
